### PR TITLE
IOS #29 Implement top bar navigation for drawer

### LINKED ISSUE #29: Implement top bar navigation
See mockup
http://imgur.com/RJOv6vz

### PR DESCRIPTION
**Pull request by raymondjacobson**
Vendorize support for drawer controller via
https://github.com/whereuat/SlideMenuControllerSwift

Closes #29

### CODE REVIEW

File: whereuat-ios/AppDelegate.swift [hunk 1]
```diff
@@ -57,7 +58,15 @@ class AppDelegate: UIResponder, UIApplicationDelegate, GGLInstanceIDDelegate, GC
...
         // Check if we are registered already. If so, instantiate ContactsViewController. Otherwise present RegisterViewController
         if (isRegistered) {
-            initialViewController = storyBoard.instantiateViewControllerWithIdentifier("ContactsViewController") as! ContactsViewController
+            // Instantiate view controllers for main views
+            let contactsViewController = storyBoard.instantiateViewControllerWithIdentifier("ContactsViewController") as! ContactsViewController
+            let leftViewController = UIViewController()
+            
+            // Instantiate navigation bar view
+            let nvc: UINavigationController = UINavigationController(rootViewController: contactsViewController)
+//            leftViewController.contactsViewController = nvc
```
Comment (kangp3) [inline whereuat-ios/AppDelegate.swift]: Delete this comment

File: whereuat-ios/Extension/UIViewControllerExtension.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    /*
+     * setNavigationBarItems removes the icons in the navigation bar
+     */
+    func removeNavigationBarItems() {
+        self.navigationItem.leftBarButtonItems = nil
+        self.navigationItem.rightBarButtonItems = nil
+        self.slideMenuController()?.removeLeftGestures()
+        self.slideMenuController()?.removeRightGestures()
+    }
+}
```
Comment (kangp3) [inline whereuat-ios/Extension/UIViewControllerExtension.swift]: newline

File: whereuat-ios/Extension/UIViewControllerExtension.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            pageDescription.setTitleTextAttributes([NSFontAttributeName: font], forState: UIControlState.Normal)
+        }
+        self.navigationItem.leftBarButtonItems = [drawerButton, pageDescription]
+
+        self.slideMenuController()?.removeLeftGestures()
+        self.slideMenuController()?.removeRightGestures()
+        self.slideMenuController()?.addLeftGestures()
+        self.slideMenuController()?.addRightGestures()
+    }
+
```
Comment (raymondjacobson) [inline whereuat-ios/Extension/UIViewControllerExtension.swift]: Kill this fn

**Comment by kangp3** [comment]
lgtm

**Pull request merged**